static/app/views/projectInstall/platformIntegrationSetup.tsx

```diff
@@ -1,172 +1,150 @@
 import ButtonBar from 'sentry/components/buttonBar';
-import DeprecatedAsyncComponent from 'sentry/components/deprecatedAsyncComponent';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
 import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
 import type {IntegrationProvider} from 'sentry/types/integrations';
-import type {Organization} from 'sentry/types/organization';
 import type {PlatformIntegration, Project} from 'sentry/types/project';
 import {trackAnalytics} from 'sentry/utils/analytics';
-import {browserHistory} from 'sentry/utils/browserHistory';
-import {trackIntegrationAnalytics} from 'sentry/utils/integrationUtil';
+import {useApiQuery} from 'sentry/utils/queryClient';
 import normalizeUrl from 'sentry/utils/url/normalizeUrl';
-import withOrganization from 'sentry/utils/withOrganization';
+import {useNavigate} from 'sentry/utils/useNavigate';
+import useOrganization from 'sentry/utils/useOrganization';
 import AddInstallationInstructions from 'sentry/views/onboarding/components/integrations/addInstallationInstructions';
 import PostInstallCodeSnippet from 'sentry/views/onboarding/components/integrations/postInstallCodeSnippet';
 import {PlatformDocHeader} from 'sentry/views/projectInstall/platformDocHeader';
 import {AddIntegrationButton} from 'sentry/views/settings/organizationIntegrations/addIntegrationButton';
 
 import FirstEventFooter from './components/firstEventFooter';
 
-type Props = {
+interface PlatformIntegrationSetupProps {
   integrationSlug: string;
+  loading: boolean;
   onClickManualSetup: () => void;
-  organization: Organization;
   platform: PlatformIntegration | undefined;
   project: Project | undefined;
-} & DeprecatedAsyncComponent['props'];
-
-type State = {
-  installed: boolean;
-  integrations: {providers: IntegrationProvider[]};
-} & DeprecatedAsyncComponent['state'];
-
-class PlatformIntegrationSetup extends DeprecatedAsyncComponent<Props, State> {
-  getDefaultState() {
-    return {
-      ...super.getDefaultState(),
-      installed: false,
-      integrations: {providers: []},
-    };
-  }
-
-  componentDidMount() {
-    super.componentDidMount();
-    window.scrollTo(0, 0);
+}
 
-    const {platform} = this.props;
+export function PlatformIntegrationSetup({
+  project,
+  platform,
+  onClickManualSetup,
+  integrationSlug,
+  loading,
+}: PlatformIntegrationSetupProps) {
+  const organization = useOrganization();
+  const [installed, setInstalled] = useState(false);
+  const navigate = useNavigate();
+
+  const {
+    data: integrations,
+    isPending,
+    isError,
+    refetch,
+  } = useApiQuery<{providers: IntegrationProvider[]}>(
+    [
+      `/organizations/${organization.slug}/config/integrations/?provider_key=${integrationSlug}`,
+    ],
+    {
+      enabled: !!integrationSlug,
+      staleTime: 0,
+    }
+  );
 
+  useEffect(() => {
+    window.scrollTo(0, 0);
```

Comment:
I would move the `scrollTo` into a separate `useEffect` to make sure it only runs on mount.